plugins/woocommerce/templates/single-product/product-thumbnails.php

```diff
@@ -24,6 +24,11 @@
 
 global $product;
 
+// Add defensive check for product existence
+if (!$product || !($product instanceof WC_Product)) {
+    return '';
+}
+
```

Comment:
I think this is a good change, thanks @imanish003! We should never assume data exists imo.

Just a few things, there are some linting errors and also there's no need for the parenthesis around the `instanceof` check

```suggestion
// Add defensive check for product existence.
if ( ! $product || ! $product instanceof WC_Product ) {
	return '';
}
```

Reply:
Ah! I am not sure why my IDE isn't showing these lint errors. Thanks for pointing these out.